src/components/navbar/index.cjsx

```diff
@@ -53,6 +54,9 @@ module.exports = React.createClass
   transitionToMenuItem: (routeName, params) ->
     @context.router.transitionTo(routeName, params)
 
+  shouldHide: ->
+    @state.scrollTop > 60 and document.getElementsByClassName('task')
```

Comment:
This bit breaks the exercise builder since the header hides. I think the exercise builder should probably change its margins to be similar (to regain vertical screen space).

Not pinned:

![image](https://cloud.githubusercontent.com/assets/253202/7538110/bb1a989a-f56b-11e4-8e93-aabaabd985fd.png)

Pinned:

![image](https://cloud.githubusercontent.com/assets/253202/7538118/c86d57f8-f56b-11e4-9209-4a1bfb62ef26.png)